### PR TITLE
Make translation strings collection workflow use a dedicated branch

### PR DESCRIPTION
**Pull request by webknjaz**

### CODE REVIEW

**Comment by webknjaz** [comment]
This should help us keep Weblate source strings up-to-date, finally.

File: .github/workflows/translation.yml [hunk 1]
```diff
@@ -1,47 +1,77 @@
...
-           git push --atomic origin HEAD:${{ github.event.repository.default_branch }}
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Grab the repo src
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0  # To reach the common commit
+    - name: Setup git user as [bot]
```
Comment (rffontenelle) [inline .github/workflows/translation.yml:24]: Consider the verb "Set up", as "setup" is technically noun #nitpicky

Comment (webknjaz) [reply .github/workflows/translation.yml:24]: Yeah, you're right. Currently, there's a more serious problem with this: Weblate doesn't work like Pradyun and I expected and now we need to think about how to address that...
```suggestion
    - name: Set up git user as [bot]
```

**Comment by rffontenelle** [comment]
@webknjaz you said it is not working on Weblate. Not sure what's up, but have you considered setting on Weblate the input POT to the new branch (TBH I don't know if you

**Comment by webknjaz** [comment]
> @webknjaz you said it is not working on Weblate. Not sure what's up, but have you considered setting on Weblate the input POT to the new branch?

That's exactly what I tried and it doesn't work the way I expected, it just sets a PR branch where Weblate tries to push stuff.
I need to think about it. I think moving the MO-files to a separate branch too and pointing RTD there would work.

> TBH I don't know if you can set separate branches for POT and PO. If you can't, you might want to consider a different repository for translations. See https://docs.weblate.org/en/latest/faq.html#how-can-i-limit-weblate-access-to-only-translations-without-exposing-source-code-to-it 

I wanted to avoid this.

**Comment by rffontenelle** [comment]
Consider posting the issue on [Weblate discussions](https://github.com/WeblateOrg/weblate/discussions).

**Comment by webknjaz** [comment]
> Consider posting the issue on [Weblate discussions](https://github.com/WeblateOrg/weblate/discussions).

I just realized that we can keep both the MOs and POT files in the `source/translation` branch and not `main`. Then, we'd publish the translated docs from that branch and Weblate would also send translation updates there w/o us needing to have those files in `main`. And also `source/translation` would be auto-updated from `main` continuously. So probably no need to raise this discussion anymore...

**Review by meowmeowmeowcat — APPROVED** [review]

**Review by pradyunsg — APPROVED** [review]

**Comment by webknjaz** [comment]
Turns out, there was a bug in this PR. Here's the hotfix: https://github.com/pypa/packaging.python.org/pull/1021.

**Pull request merged**